### PR TITLE
ansible-create/put-api

### PR DESCRIPTION
**Pull request by ChunxiAlexLuo**
Signed-off-by: Chunxi Luo chuluo@redhat.com

https://github.com/open-cluster-management/grc-ui/pull/602

### CODE REVIEW

**Comment by ycao56** [comment]
please add tests

File: src/v1/models/ansible.js [hunk 1]
```diff
@@ -107,4 +107,78 @@ export default class AnsibleModel extends KubeModel {
...
+      update: {
+        errors: updateErr,
+        result: updateRes,
+      },
+    };
+  }
+
+  async ansibleJobsAction(action, json) {
+    const namespace = _.get(json, 'metadata.namespace');
+    const url = `/apis/tower.ansible.com/v1alpha1/namespaces/${namespace}/ansiblejobs`;
```
Comment (ycao56) [inline src/v1/models/ansible.js:167]: this doesn't look correct to me. You shouldn't need to modify ansiblejob. The UI should only modify PolicyAutomation CR

File: src/v1/models/ansible.js [hunk 2]
```diff
@@ -107,4 +107,78 @@ export default class AnsibleModel extends KubeModel {
       job: _.get(au, 'status.k8sJob.namespacedName'),
     }));
   }
+
+  async createAndUpdateAnsibleJobs(args) {
```
Comment (ycao56) [inline src/v1/models/ansible.js:111]: should be `PolicyAutomation` insteand of `AnsibleJobsa`

File: src/v1/models/ansible.js (hunk 1 above)
Comment (ycao56) [reply src/v1/models/ansible.js:167]: The API is wrong. it should be `/apis/policy.open-cluster-management.io/v1beta1/namespaces/${namespace}/policyautomations`

File: src/v1/mocks/Ansible.js [hunk 3]
```diff
@@ -444,3 +444,47 @@ export const mockAnsibleJobListResponse = {
     selfLink: '/apis/tower.ansible.com/v1alpha1/namespaces/default/ansiblejobs',
   },
 };
+
+export const mockCreateAnsibleJobResponse = {
```
Comment (ycao56) [inline src/v1/mocks/Ansible.js:448]: rename to `policyautomation` to match the actual call

File: src/v1/mocks/Ansible.js [hunk 4]
```diff
@@ -444,3 +444,47 @@ export const mockAnsibleJobListResponse = {
...
+      name: 'Demo Job Template',
+      secret: 'grc-testing',
+      extra_vars: {
+        selector: 'target-cluster',
+      },
+    },
+  },
+};
+
+export const mockUpdateAnsibleJobResponse = {
```
Comment (ycao56) [inline src/v1/mocks/Ansible.js:470]: rename to `policyautomation` to match the actual call

File: src/v1/models/ansible.js [hunk 5]
```diff
@@ -121,4 +121,59 @@ export default class AnsibleModel extends KubeModel {
...
+    return {
+      create,
+      update,
+    };
+  }
+
+  async ansibleAutomationPromise(jsons, action) {
+    const resArray = [];
+    const errArray = [];
+    const result = await Promise.all(jsons.map((json) => this.ansibleJobsAction(json, action)
```
Comment (ycao56) [inline src/v1/models/ansible.js:138]: rename to `policyautomation` to match the actual call

File: src/v1/schema/ansible.js [hunk 6]
```diff
@@ -44,4 +44,7 @@ export const resolver = {
     copyAnsibleSecret: (parent, args, { ansibleModel }) => ansibleModel.copyAnsibleSecret(args),
     ansibleAutomationHistories: (parent, args, { ansibleModel }) => ansibleModel.ansibleAutomationHistories(args),
   },
+  Mutation: {
+    createAndUpdateAnsibleAutomation: (parent, args, { ansibleModel }) => ansibleModel.createAndUpdateAnsibleAutomation(args),
```
Comment (ycao56) [inline src/v1/schema/ansible.js:48]: rename to `policyautomation` to match the actual call

File: src/v1/schema/__snapshots__/ansible.test.js.snap [hunk 7]
```diff
@@ -243,3 +243,28 @@ Object {
...
+      "extensions": Object {
+        "code": "INTERNAL_SERVER_ERROR",
+      },
+      "locations": Array [
+        Object {
+          "column": 9,
+          "line": 3,
+        },
+      ],
+      "message": "_this.policyAutomationAction is not a function",
```
Comment (ycao56) [inline src/v1/schema/__snapshots__/ansible.test.js.snap:263]: the test doesn't work

Comment (ChunxiAlexLuo) [reply src/v1/schema/__snapshots__/ansible.test.js.snap]: @ycao56  updated unit test

**Review by ycao56 — APPROVED** [review]

**Pull request merged**